recipes/chalk-to-util-styletext/src/workflow.ts

```diff
@@ -0,0 +1,454 @@
+			rule: { kind: "named_imports" },
+		});
+
+		if (namedImports) {
+			const importSpecifiers = namedImports.findAll({
+				rule: { kind: "import_specifier" },
+			});
+
+			for (const specifier of importSpecifiers) {
+				const importedName = specifier.field("name");
+				const alias = specifier.field("alias");
+
+				if (importedName) {
+					const imported = importedName.text();
+					const local = alias ? alias.text() : imported;
+
+					names.push({ imported, local });
+				}
+			}
+		}
+	}
+	// Handle CommonJS requires: const { red, blue: foo } = require('chalk')
+	// Handle dynamic imports: const { red, blue: foo } = await import('chalk')
+	else if (statement.kind() === "variable_declarator") {
+		const nameField = statement.field("name");
+
+		if (nameField && nameField.kind() === "object_pattern") {
+			const properties = nameField.findAll({
+				rule: {
+					any: [{ kind: "shorthand_property_identifier_pattern" }, { kind: "pair_pattern" }],
+				},
+			});
+
+			for (const prop of properties) {
+				if (prop.kind() === "shorthand_property_identifier_pattern") {
+					// { red } - shorthand
+					const name = prop.text();
+
+					names.push({ imported: name, local: name });
+				} else if (prop.kind() === "pair_pattern") {
+					// { red: foo } - with alias
+					const key = prop.field("key");
+					const value = prop.field("value");
+
+					if (key && value) {
+						const imported = key.text();
+						const local = value.text();
+
+						names.push({ imported, local });
+					}
+				}
+			}
+		}
+	}
+
+	return names;
+}
+
+/**
+ * Extract the first text argument from a function call
+ */
+function getFirstCallArgument(call: SgNode<Js>): string | null {
+	const args = call.field("arguments");
+
+	if (!args) return null;
+
+	const argsList = args.children().filter((c) => {
+		const excluded = [",", "(", ")"];
+		return !excluded.includes(c.kind());
+	});
+
+	if (argsList.length === 0) return null;
+
+	return argsList[0].text();
+}
+
+/**
+ * Generate a styleText replacement for a single style
+ */
+function createStyleTextReplacement(styleMethod: string, textArg: string): string {
+	return `styleText("${styleMethod}", ${textArg})`;
+}
+
+/**
+ * Generate a styleText replacement for multiple styles
+ */
+function createMultiStyleTextReplacement(styles: string[], textArg: string): string {
+	if (styles.length === 1) {
+		return createStyleTextReplacement(styles[0], textArg);
+	}
+
+	const stylesArray = `[${styles.map((s) => `"${s}"`).join(", ")}]`;
+
+	return `styleText(${stylesArray}, ${textArg})`;
+}
+
+/**
+ * Process destructured imports and transform direct function calls
+ */
+function processDestructuredImports(
+	rootNode: SgNode<Js>,
+	destructuredNames: Array<{ imported: string; local: string }>,
+	edits: Edit[],
+): void {
+	for (const name of destructuredNames) {
+		const directCalls = rootNode.findAll({
+			rule: {
+				kind: "call_expression",
+				has: {
+					field: "function",
+					kind: "identifier",
```

Comment:
Could you check if your Biome is working well? The entire project is using single quotes